crates/gpui/src/platform/linux/wayland/client.rs

```diff
@@ -57,17 +60,27 @@ pub(crate) struct WaylandClientState {
     keyboard_focused_window: Option<Rc<WaylandWindowState>>,
 }
 
+#[derive(Clone)]
+pub(crate) struct WaylandClientState(Arc<Mutex<WaylandClientStateInner>>);
```

Comment:
nit: I think we'd prefer to not have this new type here, and instead move the `Arc<Mutext<..>>` into the WaylandClient :)